src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -6371,106 +6371,87 @@ public void addSnapshot(TableSnapshot snapshot) {
     }
 
     @Override
-    public boolean getTrackWarningsEnabled()
+    public boolean getReadThresholdsEnabled()
     {
-        return DatabaseDescriptor.getTrackWarningsEnabled();
+        return DatabaseDescriptor.getReadThresholdsEnabled();
     }
 
     @Override
-    public void setTrackWarningsEnabled(boolean value)
+    public void setReadThresholdsEnabled(boolean value)
     {
-        DatabaseDescriptor.setTrackWarningsEnabled(value);
-        logger.info("updated track_warnings.enabled to {}", value);
+        DatabaseDescriptor.setReadThresholdsEnabled(value);
     }
 
     @Override
-    public long getCoordinatorLargeReadWarnThresholdKB()
+    public String getCoordinatorLargeReadWarnThreshold()
     {
-        return DatabaseDescriptor.getCoordinatorReadSizeWarnThresholdKB();
+        return DatabaseDescriptor.getCoordinatorReadSizeWarnThreshold().toString();
     }
 
     @Override
-    public void setCoordinatorLargeReadWarnThresholdKB(long threshold)
+    public void setCoordinatorLargeReadWarnThreshold(String threshold)
     {
-        if (threshold < 0)
-            throw new IllegalArgumentException("threshold " + threshold + " is less than 0; must be positive or zero");
-        DatabaseDescriptor.setCoordinatorReadSizeWarnThresholdKB(threshold);
-        logger.info("updated track_warnings.coordinator_large_read.warn_threshold_kb to {}", threshold);
+        DatabaseDescriptor.setCoordinatorReadSizeWarnThreshold(threshold);
```

Comment:
It seems that we are doing the conversion from `String` to `DataStorageSpec` on `DatabaseDescriptor`, and the opposite conversion from `DataStorageSpec` to `String` on `StorageService`. Wouldn't it make more sense to do both conversions on the same class, probably on `StorageService` which is the one dealing with JMX?

Reply:
yes it would